### PR TITLE
OOP-School-Library add basic UI

### PR DESCRIPTION
**Pull request by 1GORDON**
# In this project, I achieved the following
- Created the main file that will serve as my console app entry-point.
- List all books and people.
- Created a book and a rental.
- Created a person(teacher or student)
- List all rentals for a given person id.
- Defined an entry point using the main method that is invoked at the end of the file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,36 +1,35 @@
...
-  def can_use_services?
-    of_age? || parent_permission == true
+  def add_rental(date, book)
+    Rental.new(date, book, self)
   end
+
+  private :of_age?
 end
 
 person = Person.new(100, 'gordontinyefuza')
```
Comment (o-rashad) [inline person.rb]: - [x] Don't forget to remove any remainings from the developing process like those statements  [35 - 38] 😊

**Review by o-rashad — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

Asslam Alikom @1GORDON 👋😄,
I'm Omar and I will be your reviewer for this project 😊

There are some issues 🐞 that you still need to work on to go to the next project but you are almost there! 🏁🏁🏁

## Required Changes ♻️
- [x] _Check the comments under the review._
- [x]  Your app is working perfectly only when I try to add a new `person` instance it came up with this error ... 
> 
As you may see the error explains itself... we are trying to pass 4 arguments while the function is only defined to accept 2 to 3 arguments.. kindly fix that and any errors may show up after fixing this one too.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
-require_relative './rental'
+require_relative 'corrector'
+require_relative 'rental'
 
 class Person
-  attr_accessor :name, :age, :rental
+  attr_accessor :name, :age, :rentals, :parent_permission
   attr_reader :id
 
   def initialize(age, name = 'Unknown', parent_permission: true)
```
Comment (dasileker) [inline person.rb]: - [x] One of your parameters in line 8 is wrong, I suggest to take a careful look to it and fix it.

File: student.rb [hunk 3]
```diff
@@ -1,14 +1,15 @@
-require_relative './person'
+require_relative 'person'
+require_relative 'classroom'
 
 class Student < Person
-  attr_accessor :classroom
+  attr_reader :classroom
 
   def initialize(_classroom, age, name = 'Unknown', parent_permission: true)
```
Comment (dasileker) [inline student.rb]: - [x] Also you have Two incorrect parameters, please fix them.

File: student.rb [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
     super(age, name, parent_permission)
     @classroom = classroom
   end
 
-  def owner=(classroom)
+  def classroom=(classroom)
     @classroom = classroom
     classroom.students.push(self) unless classroom.students.include?(self)
   end
```
Comment (dasileker) [inline student.rb:17]: - [x] I don't know why you do have a definition for the classroom. but I think it's not necessary.

**Review by dasileker — CHANGES_REQUESTED** [review]
## **Hi @1GORDON,** 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

- Please Fix all issues you still have the same error as the previews colleague said. I did take an extended look at your files codes and you have to correct all the errors and remove the unnecessary codes implemented.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
## Hi @1GORDON, 

# Highlights
- Good job implementing the required changes from the previous reviewers ✔️ 
- Your UI works as expected ✔️ 

# Status: Approved ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 

Approved emoji

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**